projects/RabbitMQ.Client/client/impl/SslHelper.cs

```diff
@@ -120,16 +120,16 @@ await TryAuthenticating(options)
                     }
                 }
             }
-            if (localCertificates != null && localCertificates.Count > 0)
+            if (localCertificates.Count > 0)
             {
                 return localCertificates[0];
             }
 
-            return null;
+            return null!;
```

Comment:
If this is a valid scenario, the method return type should be `X509Certificate?`. If this is not a valid scenario, and exception should be thrown instead of returning null.

Reply:
We can't change the definition, as it's from the framework. I didn't touch the original method's way. @lukebakken suppress or throwing

Reply:
It seems that there is missing a nullable annotation in the framework for the result of LocalCertificateSelectionCallback, because it is checked where it is used:
- https://source.dot.net/#System.Net.Security/System/Net/Security/SslStream.Protocol.cs,702
- https://source.dot.net/#System.Net.Security/System/Net/Security/SslStream.Protocol.cs,355

Reply:
@lechu445 we are constrained by the type signature of the [`LocalCertificateSelectionCallback` delegate](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.localcertificateselectioncallback?view=netstandard-2.0&devlangs=csharp&f1url=%3FappId%3DDev17IDEF1%26l%3DEN-US%26k%3Dk(System.Net.Security.LocalCertificateSelectionCallback)%3Bk(DevLang-csharp)%26rd%3Dtrue).

Reply:
That looks like an error in the annotation of the [LocalCertificateSelectionCallback delegate](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.localcertificateselectioncallback?view=netstandard-2.0&devlangs=csharp&f1url=%3FappId%3DDev17IDEF1%26l%3DEN-US%26k%3Dk(System.Net.Security.LocalCertificateSelectionCallback)%3Bk(DevLang-csharp)%26rd%3Dtrue).

Reply:
Does anyone feel like reporting this in the right github.com/dotnet repo?

Reply:
@paulomorgado that was my first thought too. Then I realized that in https://source.dot.net/#System.Net.Security/System/Net/Security/SslStream.Protocol.cs,702 there is thrown a NotSupportedException in case of null was returned.